dbt/adapters/databricks/connections.py

```diff
@@ -1255,6 +1282,81 @@ def exponential_backoff(attempt: int) -> int:
             retry_timeout=(timeout if timeout is not None else exponential_backoff),
         )
 
+    @classmethod
+    def _open2(cls, connection: Connection) -> Connection:
```

Comment:
We could put these into separate namespaces (if we continue functional) or separate class (if we want to be OOP) so that they could both be called open, and then pass the appropriate function in when we create the connection object.